studio/pages/_app.tsx

```diff
@@ -78,32 +79,51 @@ function CustomApp({ Component, pageProps }: AppPropsWithLayout) {
 
   const getSavingState = () => rootStore.content.savingState
 
-  // prompt the user if they try and leave with unsaved content store changes
+  const onAcceptConsent = () => {
+    if (typeof window !== 'undefined')
+      localStorage.setItem(LOCAL_STORAGE_KEYS.TELEMETRY_CONSENT, 'true')
+    if (consentToastId.current) toast.dismiss(consentToastId.current)
+  }
+  const onOptOut = () => {
+    if (typeof window !== 'undefined')
+      localStorage.setItem(LOCAL_STORAGE_KEYS.TELEMETRY_CONSENT, 'false')
+    if (consentToastId.current) toast.dismiss(consentToastId.current)
+  }
```

Comment:
small optimisation: can move these function definitions inside the useEffect to save reallocating them every render

Reply:
oooh yeah this makes sense gotcha